src/main/java/duke/command/MainUI.java

```diff
@@ -0,0 +1,149 @@
+                Deadline d = new Deadline(deadlineDescription,dueDate);
+                if (isDone){
+                    d.markAsDone();
+                }
+                taskArrayList.add(d);
+                break;
+            case 'E':
+                int durationStartingIndex = taskSentence.indexOf("(at:");
+                String duration = taskSentence.substring(durationStartingIndex + 5, taskSentence.length()-1);
+                String eventDescription = taskSentence.substring(7,durationStartingIndex);
+//                System.out.println(durationStartingIndex+":"+duration);
+                Event e = new Event(eventDescription,duration);
+                if (isDone){
+                    e.markAsDone();
+                }
+                taskArrayList.add(e);
+                break;
+            default:
+                break;
+            }
+        }
+        System.out.println("Done loading the data!");
+        printDivider();
+
+    }
+
+    private static void runProgram(Scanner in) {
+        try {
+            File data = new File("data/duke.txt");
+            if (data.createNewFile()) {
+                System.out.println("Your file that stores all your tasks is not found! (data/duke.txt) :(");
+                System.out.println("Creating new file now to store your future tasks! :)");
+                System.out.println("File created: " + data.getName());
+                printDivider();
+            } else {
+                loadData();
+            }
+
+        } catch (IOException e) {
+            System.out.println("We couldn't find the 'data' folder in your directory. This data folder will store your tasks.");
+            System.out.println("Creating 'data' folder now...");
+            File relativePathFileName = new File(DIRECTORY_NAME);
+            String fullPath = relativePathFileName.getAbsolutePath();
+            File absolutePathFileName = new File(fullPath);
+            boolean bool = absolutePathFileName.mkdirs();
+            if(bool) {
+                System.out.println("'data' folder created successfully");
+            } else {
+                System.out.println("Sorry, couldn't create 'data' folder in your directory");
+            }
+        }
+        printWelcomeMessage();
+        while (!isEnding){
+            String input = in.nextLine();
+            switch (input){
+            case LIST_COMMAND:
+                TaskManager.printAllTasks(taskArrayList);
+                break;
+            case BYE_COMMAND:
+                stopProgram();
+                break;
+            default:
+                try {
+                    TaskManager.handleTask(input);
+                } catch (CommandNotFoundException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            }
+        }
+    }
+
+    public static void printWelcomeMessage() {
+        String logo = " ____        _        \n"
```

Comment:
Can consider declaring this as a constant as the logo does not change.

Reply:
Thanks, I have made the changes :)